### PR TITLE
chore: add initial doc for yeti branching strategy

### PR DESCRIPTION
**Pull request by will-t-harris**
Please feel free to leave any comments, or point out anything that I've missed! 👍

### CODE REVIEW

**Review by Cscharf91 — APPROVED** [review]

File: docs/yeti-branching-strategy.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+```bash
+git checkout -b feat/add-widget-generation-service
+```
+
+When the feature is ready for review we'd open a pull request against the `develop` branch. Once the feature is approved it is **squashed and merged** into `develop` and automatically deployed to staging.  
+
+## How Do We Deploy to Production?
+
+Deploying to production should be fairly simple. All we need to do is merge `develop` into `main` and the production deployment should start automatically.
```
Comment (rmutter) [inline docs/yeti-branching-strategy.md:35]: Is it worth documenting our merge strategy here? Merge commit, rebase, etc. We've also had some accidental squash and merges in the past from `develop` to `main`.

**Review by rmutter — APPROVED** [review]

**Review by pfuscoe — APPROVED** [review]
We're still experimenting with the ClickUp integration for GitHub, but we might want to update the naming section to include a note on that once we've ironed out the kinks.

File: docs/yeti-branching-strategy.md (hunk 1 above)
Comment (will-t-harris) [reply docs/yeti-branching-strategy.md:35]: This is a good callout, thanks. I'll add to this tomorrow. 👍

**Review by dgillen27 — APPROVED** [review]
🌞

**Review by jonnyschult — APPROVED** [review]

**Review by emcclosky — APPROVED** [review]

**Pull request merged**